plaidml/bridge/openvino/ops/prior_box.cpp

```diff
@@ -0,0 +1,383 @@
+
+        if (min_element_size <= max_element_size) {
+          C_out = op::concatenate({C_min_size, C_max_size}, -1);
+        } else {
+          C_out = op::concatenate(
+              {edsl::reshape(
+                   op::concatenate(
+                       {op::slice(C_min_size).add_dim(0, H).add_dim(0, W).add_dim(0, result_size).add_dim(0, 4),
+                        C_max_size},
+                       -1),
+                   {H, W, result_size * 8}),
+               edsl::reshape(op::slice(C_min_size)
+                                 .add_dim(0, H)
+                                 .add_dim(0, W)
+                                 .add_dim(result_size, min_element_size)
+                                 .add_dim(0, 4),
+                             {H, W, (min_element_size - result_size) * 4})},
+              -1);
+        }
+      }
+
+      // aspect_ratio
+      if (aspect_ratios_scale_size.size() > 0) {
+        std::vector<float> min_size_ar;
+        if (attrs.scale_all_sizes) {
+          for (auto ms : min_size) min_size_ar.push_back(ms);
+        } else {
+          min_size_ar.push_back(min_size[0]);
+        }
+        std::vector<float> box_ar;
+        for (auto ms : min_size_ar) {
+          for (auto ar : aspect_ratios_scale_size) {
+            float box_width = ms * 0.5f * ar;
+            float box_height = ms * 0.5f / ar;
+            float first = -box_width / IW;
+            float second = -box_height / IH;
+            box_ar.push_back(first);
+            box_ar.push_back(second);
+            box_ar.push_back(-first);
+            box_ar.push_back(-second);
+          }
+        }
+        int msa_size = min_size_ar.size();
+        int ar_size = aspect_ratios_scale_size.size();
+        TensorShape shape(DType::FLOAT32, {msa_size * ar_size, 4});
+        Buffer buffer_ar(shape);
+        buffer_ar.copy_from(box_ar.data());
+        auto B_ar = edsl::Constant(buffer_ar, "box_ar");
+
+        auto C_ar = op::repeat(op::unsqueeze(C_mix_box, {-2})).count(msa_size * ar_size).axis(-2) + B_ar;
+        if (attrs.scale_all_sizes) {
+          auto C_ar_reshape = edsl::reshape(C_ar, {H, W, min_element_size, 4 * ar_size});
+          if (min_element_size <= max_element_size) {
+            C_out = op::concatenate({C_min_size, C_max_size, C_ar_reshape}, -1);
+          } else {
+            C_out = op::concatenate(
+                {edsl::reshape(
+                     op::concatenate(
+                         {op::slice(C_min_size).add_dim(0, H).add_dim(0, W).add_dim(0, result_size).add_dim(0, 4),
+                          C_max_size,
+                          op::slice(C_ar_reshape)
+                              .add_dim(0, H)
+                              .add_dim(0, W)
+                              .add_dim(0, result_size)
+                              .add_dim(0, 4 * ar_size)},
+                         -1),
+                     {H, W, result_size * (8 + 4 * ar_size)}),
```

Comment:
generally we should avoid using magic numbers like 8 or 4,  either define it as a constant variable or avoid using them

Reply:
In this op, most numbers are infered naturally from shapes of operands, such as combine two x*1 to get x*2,  two x*2 to x*4,   it seems we do not have one name to cover these constants as they are just a result of calculation.

Reply:
I see. I guess we can keep it as it is.

Reply:
May be I can change some key node  and add some comments there， this may have some effect

Reply:
Yes, that would be good.  Probably a better way would be to add more variables with self-explainary names, here the call stack is too deep. Consider something like
auto job1=op::slice(...);
auto job2=op::slice(...);
auto stage1=op::concatenate(job1, job2);
auto stage2=op::concatenate(job3, job4);
reshaped_result=op::reshape(...);

Use some explainary description for the var name "job1" "stage1". If we can arrage the code to be self-explainary, that would be much better.

Reply:
If it is still too complex, then we can add some words for the readers and reviewers to understand.

Reply:
The number is ok for now.

Reply:
In fact, the first version is done with this names, but the problem here are  some vals are only used once,  the name are complexed and I wonder if combine these things can make compiler easier to do optimization, may be add some comments?

Reply:
I see. Do not worry if some vals are used only once, their names are just a way of comments. If you prefer add separate comments, that would be good as well.

Reply:
Add some comments